From 9d17be68909b2edecf4fee15a3a955f6eb50be69 Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 18:59:45 -0400
Subject: [fs] btrfs: remove unnecessary transaction commit before send

Message-id: <1407179109-13714-99-git-send-email-sandeen@redhat.com>
Patchwork-id: 87057
O-Subject: [RHEL7.1 PATCH 098/422] Btrfs: remove unnecessary transaction commit before send
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Wang Shilong <wangsl.fnst@cn.fujitsu.com>

commit 8e56338d7d0ee38ecae86d35dae43020356acca1
Author: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
Date:   Tue Jan 7 17:26:57 2014 +0800

    Btrfs: remove unnecessary transaction commit before send

    We will finish orphan cleanups during snapshot, so we don't
    have to commit transaction here.

    Signed-off-by: Wang Shilong <wangsl.fnst@cn.fujitsu.com>
    Reviewed-by: Miao Xie <miaox@cn.fujitsu.com>
    Signed-off-by: Josef Bacik <jbacik@fb.com>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/send.c b/fs/btrfs/send.c
index 4e2461b..591063d 100644
--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -4776,35 +4776,6 @@ long btrfs_ioctl_send(struct file *mnt_file, void __user *arg_)
  WARN_ON(send_root->orphan_cleanup_state != ORPHAN_CLEANUP_DONE);
 
  /*
-  * If we just created this root we need to make sure that the orphan
-  * cleanup has been done and committed since we search the commit root,
-  * so check its commit root transid with our otransid and if they match
-  * commit the transaction to make sure everything is updated.
-  */
- down_read(&send_root->fs_info->extent_commit_sem);
- if (btrfs_header_generation(send_root->commit_root) ==
-     btrfs_root_otransid(&send_root->root_item)) {
-  struct btrfs_trans_handle *trans;
-
-  up_read(&send_root->fs_info->extent_commit_sem);
-
-  trans = btrfs_attach_transaction_barrier(send_root);
-  if (IS_ERR(trans)) {
-   if (PTR_ERR(trans) != -ENOENT) {
-    ret = PTR_ERR(trans);
-    goto out;
-   }
-   /* ENOENT means theres no transaction */
-  } else {
-   ret = btrfs_commit_transaction(trans, send_root);
-   if (ret)
-    goto out;
-  }
- } else {
-  up_read(&send_root->fs_info->extent_commit_sem);
- }
-
- /*
   * Userspace tools do the checks and warn the user if it's
   * not RO.
   */
-- 
1.7.1